artifactory/services/utils/artifactoryutils.go

```diff
@@ -304,21 +304,23 @@ func loadMissingProperties(reader *content.ContentReader, readerWithProps *conte
 	for resultItem := new(ResultItem); reader.NextRecord(resultItem) == nil; resultItem = new(ResultItem) {
 		// Save the item in a buffer.
 		buffer[resultItem.GetItemRelativePath()] = resultItem
+		writeOrder = append(writeOrder, resultItem)
```

Comment:
It's worth describing here the role of ```writeOrder```. Should make it easier to read the code.